dashboard/static/js/node.js

```diff
@@ -0,0 +1,293 @@
+                                type: 'post',
+                                cache: false,
+                                data: {
+                                    id: id
+                                },
+                                dataType: 'json',
+                                success: function (result) {
+                                    if (result.success) {
+                                        //重新渲染规则
+                                        _this.renderTable(result.data); //渲染table
+                                        _this.data.nodes = result.data.nodes; //重新设置数据
+
+                                        return true;
+                                    } else {
+                                        L.Common.showErrorTip("提示", result.msg || "删除节点发生错误");
+                                        return false;
+                                    }
+                                },
+                                error: function () {
+                                    L.Common.showErrorTip("提示", "删除节点请求发生异常");
+                                    return false;
+                                }
+                            });
+                        }
+                    }]
+                });
+
+                d.show();
+            });
+
+            $(document).on("click", ".edit-node-btn", function () {
+                var tpl = $("#edit-node-tpl").html();
+                var id = $(this).attr("data-id");
+                var name = $(this).attr("data-name");
+                var ip = $(this).attr("data-ip");
+                var port = $(this).attr("data-port");
+                var api_username = $(this).attr("data-api_username");
+                var api_password = $(this).attr('data-api_password');
+
+                var html = juicer(tpl, {
+                    n: {
+                        id: id,
+                        name: name,
+                        ip: ip,
+                        port: port,
+                        api_username: api_username,
+                        api_password: api_password
+                    }
+                });
+
+                var d = dialog({
+                    title: "编辑节点",
+                    width: 680,
+                    content: html,
+                    modal: true,
+                    button: [{
+                        value: '取消'
+                    }, {
+                        value: '保存修改',
+                        autofocus: false,
+                        callback: function () {
+
+                            var name = $("input[name=name]").val();
+                            var ip = $("input[name=ip]").val();
+                            var port = $("input[name=port]").val();
+                            var api_username = $("input[name=api_username]").val();
+                            var api_password = $("input[name=api_password]").val();
+
+                            var pattern = /^[A-Za-z][A-Za-z0-9_]+$/;
```

Comment:
@sumory The validation code seems duplicated, should we use a jQuery Validator Plugin?

Reply:
sure